project-minimization-plugin/src/main/kotlin/org/plan/research/minimization/plugin/settings/MinimizationPluginState.kt

```diff
@@ -1,39 +1,22 @@
 package org.plan.research.minimization.plugin.settings
 
-import org.plan.research.minimization.plugin.model.FileLevelStage
-import org.plan.research.minimization.plugin.model.FunctionLevelStage
 import org.plan.research.minimization.plugin.model.MinimizationStage
 import org.plan.research.minimization.plugin.model.state.*
 
-import com.intellij.openapi.components.BaseState
 import com.intellij.util.xmlb.annotations.Property
 import com.intellij.util.xmlb.annotations.XCollection
 
-class MinimizationPluginState : BaseState() {
-    var currentCompilationStrategy by enum<CompilationStrategy>(CompilationStrategy.GRADLE_IDEA)
-
-    /**
-     * A location for cloned projects
-     */
-    var temporaryProjectLocation by string("minimization-project-snapshots")
-    var snapshotStrategy by enum<SnapshotStrategy>(SnapshotStrategy.PROJECT_CLONING)
+data class MinimizationPluginState(val state: AppSettings) {
```

Comment:
Now, let's merge `MinimizationPluginState` with `AppSettings.State` and `MinimizationPluginSettings` with `AppSettings`. They just duplicate each other. The result after it is a deletion of `AppSettings` and its nested class.